### PR TITLE
Clientservice message dispatcher

### LINKED ISSUE #23: Message Receival doesn't seem to work properly
1. Connect websocket#1 (client ID A)
2. Connect websocket#2 (client ID B)
3. Send message from websocket#1 to client B
4. Message received, works
5. Disconnect websocket#1, reconnect websocket#3 (client ID C)
6. Send message from websocket#3 to client B
7. Message discarded (expecting C but was B)

As a theory (!!), it seems like `objectDataReply` only sends to the **last** registered listener

Note, that even on step 4, there's no "message discarded" log entry, even though A should've gotten and discarded it. In my theory, it would've been sent straight to B.

### PR DESCRIPTION
**Pull request by lroellin**
Based on #22 (first new commit is 99fd47e)
Would probably fix #23

### CODE REVIEW

**Comment by lroellin** [comment]
This makes the Client Service a message dispatcher. If my theory is right and `objectDataReply` can only hold one listener, this seems like a viable solution to me.

The only thing I don't like, is that I now need a back-reference to `ClientService` and it's too late in the evening to think about the implications of that. So, this is basically a working proposal!

File: src/main/kotlin/ch/hsr/dsl/dwrtc/signaling/ClientService.kt [hunk 1]
```diff
@@ -58,4 +62,29 @@ class ClientService() {
...
     }
+
+    internal fun addDirectMessageListener(sessionId: String, emitter: (ExternalClient, MessageDto) -> Unit) {
+        emitterMap[sessionId] = emitter
+
+        peer.peer().objectDataReply { senderPeerAddress, messageDto ->
+            logger.info { "got message $messageDto" }
+            if (messageDto is MessageDto) {
+                val recipientSessionId = messageDto.recipientSessionId
+                if (emitterMap.containsKey(recipientSessionId)) {
```
Comment (thde) [inline src/main/kotlin/ch/hsr/dsl/dwrtc/signaling/ClientService.kt]: `emitterMap[recipientSessionId] ?.let { it.` ?

File: src/main/kotlin/ch/hsr/dsl/dwrtc/signaling/ClientService.kt [hunk 2]
```diff
@@ -58,4 +62,29 @@ class ClientService() {
...
+                    logger.info { "message accepted, found emitter for $sessionId" }
+                    emitter(
+                            ExternalClient(
```
Comment (thde) [inline src/main/kotlin/ch/hsr/dsl/dwrtc/signaling/ClientService.kt]:
```
emitter(
     ExternalClient(messageDto.senderSessionId,senderPeerAddress), 
     messageDto
)
```

File: src/main/kotlin/ch/hsr/dsl/dwrtc/signaling/ClientService.kt [hunk 3]
```diff
@@ -58,4 +62,29 @@ class ClientService() {
...
         } else throw ClientNotFoundException("No peer found under session ID $sessionId")
     }
+
+    internal fun addDirectMessageListener(sessionId: String, emitter: (ExternalClient, MessageDto) -> Unit) {
+        emitterMap[sessionId] = emitter
+
+        peer.peer().objectDataReply { senderPeerAddress, messageDto ->
+            logger.info { "got message $messageDto" }
+            if (messageDto is MessageDto) {
+                val recipientSessionId = messageDto.recipientSessionId
```
Comment (thde) [inline src/main/kotlin/ch/hsr/dsl/dwrtc/signaling/ClientService.kt]: Maybe seperate from here

**Pull request merged**